internal/agent/metrics/accumulator.go

```diff
@@ -23,7 +36,7 @@ func NewMetricAccumulator(id string, mtype string) *MetricAccumulator {
 
 func (ma *MetricAccumulator) AccumulateCounter(inc int64) error {
 	if ma.MType != model.Counter {
-		return model.ErrMethodNotSupported
+		return model.ErrIncorrectAccess
```

Comment:
В типизированные ошибки тоже можно добавлять контекст:

```suggestion
		return fmt.Errorf("%w: expected counter, got %v", model.ErrIncorrectAccess, ma.MType)
```